### ISSUE TITLE
[Bug] wtscale is lost after saving model due to destructive .pop() in patch_scale_key

### ISSUE THREAD

**Issue by ultranationalism** [nunchaku-tech/nunchaku#862]
### Checklist

- [x] 1. I have searched for related issues and FAQs (https://nunchaku.tech/docs/nunchaku/faq/faq.html) but was unable to find a solution.
- [x] 2. The issue persists in the latest version.
- [x] 3. Please note that without environment information and a minimal reproducible example, it will be difficult for us to reproduce and address the issue, which may delay our response.
- [ ] 4. If your report is a question rather than a bug, please submit it as a discussion at https://github.com/mit-han-lab/nunchaku/discussions/new/choose. Otherwise, this issue will be closed.
- [ ] 5. If this is related to ComfyUI, please report it at https://github.com/mit-han-lab/ComfyUI-nunchaku/issues.
- [x] 6. I will do my best to describe the issue in English.

### Describe the Bug

There is a critical inconsistency between model loading and saving in Nunchaku. When a quantized model (e.g., ZImageTransformer2DModel in nvfp4 mode) is loaded and subsequently saved using standard PyTorch or Diffusers methods (e.g., pipe.save() or pipe.push(), the resulting checkpoint is incompatible with the original loader.

Core reasons found in the source code:

wtscale Loss: In nunchaku/models/transformers/utils.py, the patch_scale_key function uses state_dict_from_checkpoint.pop(f"{n}.wtscale", 1.0) to load the global weight scale. However, in SVDQW4A4Linear, wtscale is initialized as a plain Python float attribute (or a scalar tensor) but is not registered as a nn.Parameter or nn.Buffer. As a result, it is excluded from the model's state_dict() during saving.

wcscales Inconsistency: In nvfp4 mode, if LoRA is fused or specific quantization logic is triggered, wcscales (channel-wise scales) are recalculated or modified. The saved checkpoint often contains these "calculated" values, but without the corresponding wtscale, the numerical integrity is broken.

This behavior makes it impossible to re-upload or share models after putting them into a pipeline, as the essential scaling metadata disappears.

### Environment

System Specs: Ubuntu 24.04

Python Version: 3.12+

PyTorch Version: 2.9+

CUDA Version: 12.8+

Nunchaku Version: Latest main branch

### Reproduction Steps
```
import torch
import os
from nunchaku import NunchakuZImageTransformer2DModel
from diffusers import ZImagePipeline
from diffusers import ZImagePipeline, FlowMatchEulerDiscreteScheduler, AutoencoderKL
from transformers import Qwen2Tokenizer, Qwen3Model
model_id = "Tongyi-MAI/Z-Image-Turbo"

# 1. 手动加载所有组件
transformer = NunchakuZImageTransformer2DModel.from_pretrained(
    "nunchaku-tech/nunchaku-z-image-turbo/svdq-fp4_r128-z-image-turbo.safetensors"
)
vae = AutoencoderKL.from_pretrained(model_id, subfolder="vae",torch_dtype=torch.float16)
scheduler = FlowMatchEulerDiscreteScheduler.from_pretrained(model_id, subfolder="scheduler")
tokenizer = Qwen2Tokenizer.from_pretrained(model_id, subfolder="tokenizer")
text_encoder = Qwen3Model.from_pretrained(model_id, subfolder="text_encoder",dtype=torch.bfloat16)

# 2. 直接调用构造函数，完全跳过 from_pretrained 的类型检查
pipe = ZImagePipeline(
    vae=vae,
    text_encoder=text_encoder,
    tokenizer=tokenizer,
    transformer=transformer,
    scheduler=scheduler
)
pipe.push()
# or
pipe.save("path_to_save_directory")
```
Original (with wtscale): [nunchaku-tech/nunchaku-z-image-turbo](https://huggingface.co/nunchaku-tech/nunchaku-z-image-turbo/blob/main/svdq-fp4_r128-z-image-turbo.safetensors)

Saved Version (missing wtscale): [ultranationalism/nunchaku-z-image-turbo](https://huggingface.co/ultranationalism/nunchaku-z-image-turbo/blob/main/transformer/diffusion_pytorch_model.safetensors)

**Comment by hhhFuture** [comment]
> ### 检查清单
> * [x] 1. 我已搜索过相关问题和常见问题解答（https://nunchaku.tech/docs/nunchaku/faq/faq.html），但未能找到解决方案。[x] 2. 最新版本中仍然存在此问题。[x] 3. 请注意，如果没有环境信息和最小可复现示例，我们将很难重现和解决问题，这可能会延迟我们的回复。[ ] 4. 如果您要报告的是一个问题而不是一个错误，请将其作为讨论提交到https://github.com/mit-han-lab/nunchaku/discussions/new/choose。否则，此问题将被关闭。[ ] [5. 如果这与 ComfyUI 有关，请在https://github.com/mit-han-lab/ComfyUI-nunchaku/issues](https://github.com/mit-han-lab/ComfyUI-nunchaku/issues)上报告。[x] 6. 我会尽力用英语描述这个问题。
> 
> ### 描述错误
> Nunchaku 中模型加载和保存之间存在严重的不一致性。当加载量化模型（例如，nvfp4 模式下的 ZImageTransformer2DModel）并随后使用标准的 PyTorch 或 Diffusers 方法（例如，pipe.save() 或 pipe.push()）保存时，生成的检查点与原始加载器不兼容。
> 
> 从源代码中发现的核心原因：
> 
> wtscale 损失：在 nunchaku/models/transformers/utils.py 中，patch_scale_key 函数使用 state_dict_from_checkpoint.pop(f"{n}.wtscale", 1.0) 加载全局权重缩放比例。然而，在 SVDQW4A4Linear 中，wtscale 被初始化为一个普通的 Python 浮点属性（或标量张量），但并未注册为 nn.Parameter 或 nn.Buffer。因此，在保存模型时，它会被排除在 state_dict() 之外。
> 
> wcscales 不一致：在 nvfp4 模式下，如果 LoRA 融合或触发特定的量化逻辑，则 wcscales（通道级缩放）会被重新计算或修改。保存的检查点通常包含这些“计算”值，但由于缺少相应的 wtscale，数值完整性会被破坏。
> 
> 这种行为使得将模型放入管道后无法重新上传或共享模型，因为必要的缩放元数据会消失。
> 
> ### 环境
> 系统规格：Ubuntu 24.04
> 
> Python 版本：3.12+
> 
> PyTorch 版本：2.9+
> 
> CUDA 版本：12.8+
> 
> Nunchaku 版本：最新主分支
> 
> ### 重现步骤
> ```
> import torch
> import os
> from nunchaku import NunchakuZImageTransformer2DModel
> from diffusers import ZImagePipeline
> from diffusers import ZImagePipeline, FlowMatchEulerDiscreteScheduler, AutoencoderKL
> from transformers import Qwen2Tokenizer, Qwen3Model
> model_id = "Tongyi-MAI/Z-Image-Turbo"
> 
> # 1. 手动加载所有组件
> transformer = NunchakuZImageTransformer2DModel.from_pretrained(
>     "nunchaku-tech/nunchaku-z-image-turbo/svdq-fp4_r128-z-image-turbo.safetensors"
> )
> vae = AutoencoderKL.from_pretrained(model_id, subfolder="vae",torch_dtype=torch.float16)
> scheduler = FlowMatchEulerDiscreteScheduler.from_pretrained(model_id, subfolder="scheduler")
> tokenizer = Qwen2Tokenizer.from_pretrained(model_id, subfolder="tokenizer")
> text_encoder = Qwen3Model.from_pretrained(model_id, subfolder="text_encoder",dtype=torch.bfloat16)
> 
> # 2. 直接调用构造函数，完全跳过 from_pretrained 的类型检查
> pipe = ZImagePipeline(
>     vae=vae,
>     text_encoder=text_encoder,
>     tokenizer=tokenizer,
>     transformer=transformer,
>     scheduler=scheduler
> )
> pipe.push()
> # or
> pipe.save("path_to_save_directory")
> ```
> 
> 原始版本（使用 wtscale）：[nunchaku-tech/nunchaku-z-image-turbo](https://huggingface.co/nunchaku-tech/nunchaku-z-image-turbo/blob/main/svdq-fp4_r128-z-image-turbo.safetensors)
> 
> 已保存版本（缺少 wtscale）：[ultranationalism/nunchaku-z-image-turbo](https://huggingface.co/ultranationalism/nunchaku-z-image-turbo/blob/main/transformer/diffusion_pytorch_model.safetensors)

I think the problem I'm facing might be the same as yours. The corresponding issue is:https://github.com/nunchaku-tech/nunchaku/issues/861#issue-3775762775 So far, I haven't found a solution yet.

**Comment by ultism** [comment]
> > ### 检查清单
> > 
> > 
> > * [x]  1. 我已搜索过相关问题和常见问题解答（[https://nunchaku.tech/docs/nunchaku/faq/faq.html），但未能找到解决方案。[x]](https://nunchaku.tech/docs/nunchaku/faq/faq.html%EF%BC%89%EF%BC%8C%E4%BD%86%E6%9C%AA%E8%83%BD%E6%89%BE%E5%88%B0%E8%A7%A3%E5%86%B3%E6%96%B9%E6%A1%88%E3%80%82%5Bx%5D) 2. 最新版本中仍然存在此问题。[x] 3. 请注意，如果没有环境信息和最小可复现示例，我们将很难重现和解决问题，这可能会延迟我们的回复。[ ] 4. 如果您要报告的是一个问题而不是一个错误，请将其作为讨论提交到[https://github.com/mit-han-lab/nunchaku/discussions/new/choose。否则，此问题将被关闭。[](https://github.com/mit-han-lab/nunchaku/discussions/new/choose%E3%80%82%E5%90%A6%E5%88%99%EF%BC%8C%E6%AD%A4%E9%97%AE%E9%A2%98%E5%B0%86%E8%A2%AB%E5%85%B3%E9%97%AD%E3%80%82%5B) ] [5. 如果这与 ComfyUI 有关，请在https://github.com/mit-han-lab/ComfyUI-nunchaku/issues](https://github.com/mit-han-lab/ComfyUI-nunchaku/issues)上报告。[x] 6. 我会尽力用英语描述这个问题。
> > ### 描述错误
> > Nunchaku 中模型加载和保存之间存在严重的不一致性。当加载量化模型（例如，nvfp4 模式下的 ZImageTransformer2DModel）并随后使用标准的 PyTorch 或 Diffusers 方法（例如，pipe.save() 或 pipe.push()）保存时，生成的检查点与原始加载器不兼容。
> > 从源代码中发现的核心原因：
> > wtscale 损失：在 nunchaku/models/transformers/utils.py 中，patch_scale_key 函数使用 state_dict_from_checkpoint.pop(f"{n}.wtscale", 1.0) 加载全局权重缩放比例。然而，在 SVDQW4A4Linear 中，wtscale 被初始化为一个普通的 Python 浮点属性（或标量张量），但并未注册为 nn.Parameter 或 nn.Buffer。因此，在保存模型时，它会被排除在 state_dict() 之外。
> > wcscales 不一致：在 nvfp4 模式下，如果 LoRA 融合或触发特定的量化逻辑，则 wcscales（通道级缩放）会被重新计算或修改。保存的检查点通常包含这些“计算”值，但由于缺少相应的 wtscale，数值完整性会被破坏。
> > 这种行为使得将模型放入管道后无法重新上传或共享模型，因为必要的缩放元数据会消失。
> > ### 环境
> > 系统规格：Ubuntu 24.04
> > Python 版本：3.12+
> > PyTorch 版本：2.9+
> > CUDA 版本：12.8+
> > Nunchaku 版本：最新主分支
> > ### 重现步骤
> > ```
> > import torch
> > import os
> > from nunchaku import NunchakuZImageTransformer2DModel
> > from diffusers import ZImagePipeline
> > from diffusers import ZImagePipeline, FlowMatchEulerDiscreteScheduler, AutoencoderKL
> > from transformers import Qwen2Tokenizer, Qwen3Model
> > model_id = "Tongyi-MAI/Z-Image-Turbo"
> > 
> > # 1. 手动加载所有组件
> > transformer = NunchakuZImageTransformer2DModel.from_pretrained(
> >     "nunchaku-tech/nunchaku-z-image-turbo/svdq-fp4_r128-z-image-turbo.safetensors"
> > )
> > vae = AutoencoderKL.from_pretrained(model_id, subfolder="vae",torch_dtype=torch.float16)
> > scheduler = FlowMatchEulerDiscreteScheduler.from_pretrained(model_id, subfolder="scheduler")
> > tokenizer = Qwen2Tokenizer.from_pretrained(model_id, subfolder="tokenizer")
> > text_encoder = Qwen3Model.from_pretrained(model_id, subfolder="text_encoder",dtype=torch.bfloat16)
> > 
> > # 2. 直接调用构造函数，完全跳过 from_pretrained 的类型检查
> > pipe = ZImagePipeline(
> >     vae=vae,
> >     text_encoder=text_encoder,
> >     tokenizer=tokenizer,
> >     transformer=transformer,
> >     scheduler=scheduler
> > )
> > pipe.push()
> > # or
> > pipe.save("path_to_save_directory")
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 原始版本（使用 wtscale）：[nunchaku-tech/nunchaku-z-image-turbo](https://huggingface.co/nunchaku-tech/nunchaku-z-image-turbo/blob/main/svdq-fp4_r128-z-image-turbo.safetensors)
> > 已保存版本（缺少 wtscale）：[ultranationalism/nunchaku-z-image-turbo](https://huggingface.co/ultranationalism/nunchaku-z-image-turbo/blob/main/transformer/diffusion_pytorch_model.safetensors)
> 
> I think the problem I'm facing might be the same as yours. The corresponding issue is:[#861 (comment)](https://github.com/nunchaku-tech/nunchaku/issues/861#issue-3775762775) So far, I haven't found a solution yet.

I will solve it if no one do,but you may wait until I finish my  work in vllm-omni
